xarray/coding/cftime_offsets.py

```diff
@@ -1254,6 +1255,54 @@ def date_range(
     )
 
 
+def _new_to_legacy_freq(freq):
+    # xarray will now always return "ME" and "QE" for MonthEnd and QuarterEnd
+    # frequencies, but older versions of pandas do not support these as
+    # frequency strings.  Until xarray's minimum pandas version is 2.2 or above,
+    # we add logic to continue using the deprecated "M" and "Q" frequency
+    # strings in these circumstances.
+
+    # TODO: remove once requiring pandas >= 2.2
+
+    if Version(pd.__version__) < Version("2.2"):
+        freq_as_offset = to_offset(freq)
```

Comment:
I think this may have been an issue in code this was adapted from, but it occurs to me if we are using xarray's cftime-specific `to_offset` function here, is it possible that we could raise an unwanted error if one passes a frequency string that is not supported for cftime, but is supported for pandas, e.g. something like `"BMS"`? 

It's probably fine to let those pass through unmodified, but we may need to add logic to work past that exception.